formats/json/commonTest/src/kotlinx/serialization/features/JsonAlternativeNamesTest.kt

```diff
@@ -26,10 +39,33 @@ class JsonAlternativeNamesTest : JsonTestBase() {
     private val inputString2 = """{"_foo":"foo"}"""
     private val json = Json { useAlternativeNames = true }
 
+    @Test
+    fun testEnumSupportsAlternativeNames() = noLegacyJs {
+        val input = """{"enumList":["VALUE_A", "someValue", "some_value", "VALUE_B"], "checkCoercion":"someValue"}"""
```

Comment:
I see from the code that top-level enums are also properly supported (e.g. `Json.decodeFromString<AlternateEnumNames>("\"someValue\"")` works properly), but it would be nice to test it as well